File: ds-core/src/main/java/eu/groeller/ds/presentation/request/exercise/CreateWorkoutRequest.java
package eu.groeller.ds.presentation.request.exercise;

import java.time.OffsetDateTime;
import java.util.List;

public record CreateWorkoutRequest(
        List<ExerciseRecordRequest> exercises,
        String type,
        OffsetDateTime startTime,
        OffsetDateTime endTime
) {
}

File: ds-compose/app/src/main/java/eu/groeller/dsui/data/model/CreateWorkoutRequest.kt
package eu.groeller.datastreamui.data.model

import ExerciseRecordRequest
import eu.groeller.datastreamui.data.serializer.OffsetDateTimeSerializer
import kotlinx.serialization.Serializable
import java.time.OffsetDateTime

@Serializable
data class CreateWorkoutRequest(
    val type: String,
    val exercises: List<CreateExerciseRequest>,
    @Serializable(with = OffsetDateTimeSerializer::class)
    val startTime: OffsetDateTime,
    @Serializable(with = OffsetDateTimeSerializer::class)
    val endTime: OffsetDateTime
)
